spark/src/main/scala/com/gu/tableversions/spark/VersionContext.scala

```diff
@@ -126,10 +127,19 @@ object SparkSupport {
 
     val partitions = table.partitionSchema.columns.map(_.name)
 
+    val versionedUri = setVersionedScheme(table.location)
+
     dataset.toDF.write
       .partitionBy(partitions: _*)
       .mode(SaveMode.Append)
       .format(table.format.name)
-      .save(VersionedFileSystem.scheme + "://" + table.location.getPath)
+      .save(versionedUri.toString)
   }
+
+  private[spark] def setVersionedScheme(underlyingUri: URI): URI =
+    new URI(VersionedFileSystem.scheme,
+            underlyingUri.getAuthority,
+            underlyingUri.getPath,
+            underlyingUri.getQuery,
+            underlyingUri.getFragment)
```

Comment:
Good fix! I think at some point it would be worth moving all our URI-manipulating code into a single place and test it thoroughly, as we do similar things in various places and it's proved to be error prone. This is absolutely fine for the time being though!

Reply:
yeah that is probably the way to go.
alternatively in some other guardian projects we use a [scala uri implementation](https://github.com/lemonlabsuk/scala-uri) that might gives us what we need if we are ok with introducing that dependency

Reply:
Interesting, I just looked at that library yesterday actually! It'd be worth considering, though it pulls in a couple of further dependencies (parboiled and shapeless) and may be overkill for what we need (changing schemas and rewriting paths). Something to ponder if we carry on with this I guess.